encodings/dict/src/compute/compare.rs

```diff
@@ -15,14 +15,12 @@ impl CompareFn<DictArray> for DictEncoding {
         // If the RHS is constant, then we just need to compare against our encoded values.
         if let Some(const_scalar) = rhs.as_constant() {
             // Ensure the other is the same length as the dictionary
-            return compare(
+            let compare_result = compare(
                 lhs.values(),
                 ConstantArray::new(const_scalar, lhs.values().len()),
                 operator,
-            )
-            .and_then(|values| DictArray::try_new(lhs.codes(), values))
-            .map(|a| a.into_array())
-            .map(Some);
+            )?;
+            return take(compare_result, lhs.codes(), TakeOptions::default()).map(Some);
```

Comment:
once we added fill_null as a compute function we were encountering bool dictinaries that were coming from here